packages/aws-rfdk/lib/deadline/lib/sep-spotfleet.ts

```diff
@@ -0,0 +1,693 @@
+
+    const blockDeviceMappings = (props.blockDevices !== undefined ?
+      synthesizeBlockDeviceMappings(this, props.blockDevices) : undefined);
+
+    const { subnets } = props.vpc.selectSubnets(props.vpcSubnets);
+    const subnetIds = subnets.map(subnet => {
+      return subnet.subnetId;
+    });
+    const subnetId = subnetIds.length != 0 ? subnetIds.join(',') : undefined;
+
+    const instanceTagsToken = this.tagsToken(ISpotFleetResourceType.INSTANCE);
+    const spotFleetRequestTagsToken = this.tagsToken(ISpotFleetResourceType.SPOT_FLEET_REQUEST);
+
+    let launchSpecifications: any[] = [];
+
+    props.instanceTypes.map(instanceType => {
+      const launchSpecification = {
+        BlockDeviceMappings: blockDeviceMappings,
+        IamInstanceProfile: {
+          Arn: iamProfile.attrArn,
+        },
+        ImageId: this.imageId,
+        KeyName: props.keyName,
+        SecurityGroups: securityGroupsToken,
+        SubnetId: subnetId,
+        TagSpecifications: instanceTagsToken,
+        UserData: userDataToken,
+        InstanceType: instanceType.toString(),
+      };
+      launchSpecifications.push(launchSpecification);
+    });
+
+    const spotFleetRequestConfiguration = {
+      AllocationStrategy: props.allocationStrategy,
+      IamFleetRole: this.iamFleetRole.roleArn,
+      LaunchSpecifications: launchSpecifications,
+      ReplaceUnhealthyInstances: true,
+      TargetCapacity: props.targetCapacity,
+      TerminateInstancesWithExpiration: true,
+      Type: 'maintain',
+      ValidUntil: props.validUntil ? props.validUntil?.date.toUTCString() : undefined,
+      TagSpecifications: spotFleetRequestTagsToken,
+    };
+
+    const sepSpotFleetRequestConfigurations = props.deadlineGroups.map(group => {
+      const sepSpotFleetRequestConfiguration = {
+        [group]: spotFleetRequestConfiguration,
+      };
+      return sepSpotFleetRequestConfiguration;
+    });
+
+    return sepSpotFleetRequestConfigurations;
+  }
+
+  private tagsToken(resourceType: ISpotFleetResourceType): IResolvable {
+    return Lazy.any({
+      produce: () => {
+        if (this.tags.hasTags()) {
+          return [{
+            ResourceType: resourceType,
+            Tags: this.tags.renderTags(),
+          }];
+        }
+        return undefined;
+      },
+    });
+  }
+
+  private validateProps(props: SEPSpotFleetProps) {
+    this.validateInstanceTypes(props.instanceTypes);
+    this.validateGroups(props.deadlineGroups);
+    this.validateArrayGroupsSyntax(props.deadlineGroups, /^(?!none$)[a-zA-Z0-9-_]+$/i, 'deadlineGroups');
+    this.validateRegion(props.deadlineRegion, /^(?!none$|all$|unrecognized$)[a-zA-Z0-9-_]+$/i);
+    this.validateBlockDevices(props.blockDevices);
+  }
+
+  private validateInstanceTypes(array: InstanceType[]) {
+    if (array.length == 0) {
+      throw new Error('SEPSpotFleet: At least one Deadline Group is required for a Spot Fleet Request Configuration');
```

Comment:
This error message looks like it needs to be updated from a copy/paste

Reply:
Fixed